platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r88/DaemonToolchainCoexistWithCurrentOptionsCrossVersionTest.groovy

```diff
@@ -31,7 +32,8 @@ class DaemonToolchainCoexistWithCurrentOptionsCrossVersionTest extends ToolingAp
     @Requires(IntegTestPreconditions.JavaHomeWithDifferentVersionAvailable)
     def "Given toolchain properties are provided then build succeeds"() {
         given:
-        def otherJvm = AvailableJavaHomes.differentVersion
+        def otherJvm = AvailableJavaHomes.getDifferentVersion { it.javaMajorVersion < 24 }
```

Comment:
Could we use https://github.com/gradle/gradle/blob/3c890746756262d3778e12eaa5155d661d7cbdf2/testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/DefaultGradleDistribution.groovy to filter for Java versions supported by the target Gradle version?

Reply:
That tooling isn't immediately available because we have to create a `DefaultGradleDistribution` instance like

```
DefaultGradleDistribution dummy = new DefaultGradleDistribution(GradleVersion("8.9"), null, null).worksWith(jvm)
```

I'll merge the current code to unblock us, then do a follow-up to refactor the functionality out of DefaultGradleDistribution (like a static function).